### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by JeTSkY1h**
Added Tests for BookService

### CODE REVIEW

File: backend/src/main/java/com/github/JeTSkY1h/user/MyUserController.java [hunk 1]
```diff
@@ -31,10 +31,11 @@ ResponseEntity<Object> giveAdmin(@RequestBody MyUser newAdmin){
 
     @GetMapping("/bookdata")
     ResponseEntity<List<BookUserData>>getBookUserData(Principal principal){
-        if(myUserService.getBookuserData(principal.getName()).isEmpty()) {
+        Optional<MyUser> user = myUserService.findByUsername(principal.getName());
+        if(user.isEmpty()) {
             return null;
         }
-        return ResponseEntity.of(myUserService.getBookuserData(principal.getName()));
+        return ResponseEntity.ok(user.get().getBookData());
```
Comment (gossie) [inline backend/src/main/java/com/github/JeTSkY1h/user/MyUserController.java]: Why did you change that from `ResponseEntity.of` to `ResponseEntity.ok`? I would remove the `return null` and rewrite the methode like this:
```java
return ResponseEntity.of(myUserService.findByUsername(principal.getName())
     .map(user -> user.getBookData()));
```
That way you'll receive a 200 with the book data, if the user is present and a 404 if not.

**Review by gossie — DISMISSED** [review]
Just left one comment. You can change it on this PR or in another branch or not at all.

**Review by NicolaiKopka — APPROVED** [review]
